lib/datadog/di/base.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+# This file is loaded by datadog/di/init.rb.
+# It contains just the global DI reference to the (normally one and only)
+# code tracker for the current process.
+# This file should not require the rest of DI, specifically none of the
+# contrib code that is meant to be loaded after third-party libraries
+# are loaded, and also none of the rest of datadog library which also
+# has contrib code in other products.
+
+require_relative 'code_tracker'
+
+module Datadog
+  # Namespace for Datadog dynamic instrumentation.
+  #
+  # @api private
+  module DI
```

Comment:
Consider actually putting this under `DI::Base` or something like that? Specifically, I always find it mega confusing if I'm looking at `datadog/foo` and I open `datadog/foo.rb` and I'm missing code -- and I might not ever realize it.

I think as much as possible, keeping the 1:1 between namespace and file helps a lot of not getting lost in the codebase.